scene/resources/tile_set.cpp

```diff
@@ -1458,12 +1460,15 @@ Vector<Vector2> TileSet::get_tile_shape_polygon() {
 				break;
 		}
 
-		points.append(Vector2(0.0, -0.5));
-		points.append(Vector2(-0.5, overlap - 0.5));
-		points.append(Vector2(-0.5, 0.5 - overlap));
-		points.append(Vector2(0.0, 0.5));
-		points.append(Vector2(0.5, 0.5 - overlap));
-		points.append(Vector2(0.5, overlap - 0.5));
+		points.push_back(Vector2(0.0, -0.5));
+		points.push_back(Vector2(-0.5, overlap - 0.5));
+		points.push_back(Vector2(-0.5, 0.5 - overlap));
+		points.push_back(Vector2(0.0, 0.5));
+		points.push_back(Vector2(0.5, 0.5 - overlap));
+		points.push_back(Vector2(0.5, overlap - 0.5));
+	}
+
+	if (tile_shape != TileSet::TILE_SHAPE_SQUARE) {
```

Comment:
The half offset axis does not matter anymore here, but I understand it would be needed for the compatibility with the code I mentioned above (which might work in the end :thinking: ).

In any case, for cleaner code, I would simply change that to move that code inside the `else` statement above. And don't apply it to isometric shapes.

Reply:
Right, swapping xy is no longer needed for isometric tile polygon. I guess I somehow assumed not doing so could break something else, hence I've done it like that. But after rechecking it indeed seems like only `TileMap::draw_cells_outline` relies on that so it should be fine to simplify that.

Reply:
Yeah, I've just checked too, I haven't find any other place where the code relies on the order of the vertices, so that should be fine.

Reply:
Done, and tested that outlines are still drawn just fine for isometric + horizontal/vertical offset axis.